notification/webhook/sender.go

```diff
@@ -0,0 +1,101 @@
+package webhook
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/target/goalert/notification"
+)
+
+type Sender struct{}
+
+type WebhookAlert struct {
+	AlertID     int    `json:",omitempty"`
+	AlertType   string `json:",omitempty"`
+	Code        string `json:",omitempty"`
+	Summary     string `json:",omitempty"`
+	Details     string `json:",omitempty"`
+	ServiceID   string `json:",omitempty"`
+	ServiceName string `json:",omitempty"`
+	Count       int    `json:",omitempty"`
+	LogEntry    string `json:",omitempty"`
+}
+
+func NewSender(ctx context.Context) *Sender {
+	return &Sender{}
+}
+
+// Send will send an alert for the provided message type
+func (s *Sender) Send(ctx context.Context, msg notification.Message) (*notification.MessageStatus, error) {
+
+	postWithBody := func(body string) (*http.Response, error) {
+		req, err := http.NewRequest("POST", msg.Destination().Value, bytes.NewBufferString(body))
+		if err != nil {
+			return nil, err
+		}
+		req = req.WithContext(ctx)
+		req.Header.Add("Content-Type", "application/json")
+		return http.DefaultClient.Do(req)
+	}
+
+	var payload WebhookAlert
+
+	switch m := msg.(type) {
+	case notification.Test:
+		payload.AlertType = "Test"
+		payload.Summary = "Test Message"
+		payload.Details = "This is a test message from GoAlert"
+
+	case notification.Verification:
+		payload.AlertType = "Verification"
+		payload.Summary = "Verification Message"
+		payload.Details = "This is a verification message from GoAlert"
+		payload.Code = strconv.Itoa(m.Code)
+
+	case notification.Alert:
+		payload.AlertType = "Alert"
+		payload.AlertID = m.AlertID
+		payload.Summary = m.Summary
+		payload.Details = m.Details
+
+	case notification.AlertBundle:
+		payload.AlertType = "AlertBundle"
+		payload.ServiceID = m.ServiceID
+		payload.ServiceName = m.ServiceName
+		payload.Count = m.Count
+
+	case notification.AlertStatus:
+		payload.AlertType = "AlertStatus"
+		payload.AlertID = m.AlertID
+		payload.LogEntry = m.LogEntry
+
+	case notification.AlertStatusBundle:
+		payload.AlertType = "AlertStatusBundle"
+		payload.AlertID = m.AlertID
+		payload.Count = m.Count
+		payload.LogEntry = m.LogEntry
+
+	default:
+		return nil, errors.New("message type not supported")
+	}
+
+	jsonbytes, err := json.Marshal(payload)
```

Comment:
nit/convention: use `data` instead to represent a relevant set of bytes.
```suggestion
	data, err := json.Marshal(payload)
```